src/Amazon.Extensions.CognitoAuthentication/CognitoUserAuthentication.cs

```diff
@@ -68,6 +68,11 @@ public virtual async Task<AuthFlowResponse> StartWithSrpAuthAsync(InitiateSrpAut
                 initiateRequest.ClientMetadata = new Dictionary<string, string>(srpRequest.ClientMetadata);
             }
 
+            if (srpRequest.UserContextData != null)
```

Comment:
I don't think you need to explicitly check for `null` before setting `initiateRequest.UserContextData` (see similar assignment for `initiateRequest.AnalyticsMetadata`).